site/app/libraries/Utils.php

```diff
@@ -315,4 +315,32 @@ public static function stripComments(string $str): ?string {
     public static function escapeDoubleQuotes(string $str): ?string {
         return preg_replace('["]', '\"', $str);
     }
+
+    /**
+     * Transforms non-boolean values to boolean values, i.e. 'true' to true
+     * @param mixed $variable The variable that is being passed in. Type of mixed to allow for boolean, string, integer, or null.
+     */
+    public static function getBooleanValue(mixed $variable): bool {
+        // Handle variables that are already boolean
+        if (is_bool($variable)) {
+            return $variable;
+        }
+
+        if (is_string($variable)) {
+            // Handle string values peacefully, 'true' or 'on' (for javascript checkboxes),
+            $true_values = [
+                'true',
+                'on'
+            ];
+            return (in_array(strtolower(trim($variable)), $true_values, true));
+        }
+
+        if (is_int($variable)) {
+            // Handle 0 as the only false integer.
+            return $variable !== 0;
+        }
+
+        // Default to returning false
+        return false;
```

Comment:
Would it be a good idea to default to casting as a bool?  I could see a good argument for and against it.  The docs for [`boolval()`](https://www.php.net/manual/en/function.boolval.php) give some interesting ideas for test cases as well.

Reply:
I feel like false is better to be more cautious of what is returned, and specify when values should be true, but I don't have a strong argument for or against.

Reply:
I don't have a particularly strong argument for either option.  On one hand, I agree with you that simpler code is probably better, but on the other hand it's possible that differences between this function and regular PHP casting rules will be problematic in the future.  I think it's probably sufficient for now at least.